### ISSUE TITLE
Review for your JOSS submission

### ISSUE THREAD

**Issue by Het-Shah** [Planet-AI-GmbH/tfaip#3]
Hi @ChWick! I am a reviewer for your work at JOSS ([link](https://github.com/openjournals/joss-reviews/issues/3297)). Overall, the structure of the library seems great to me!  However, I had few concerns which I would like to add here 😄. I believe that these won't take a lot of time to rectify.  

Concerns regarding the repository - 
- [ ] Can you benchmark on any one scenario? For example, min/full tutorial can be benchmarked on a simple model architecture from TensorFlow. 
- [ ] Can you add more tutorials to this repository? I can find an additional repository that has different scenarios, [link](https://github.com/Planet-AI-GmbH/tfaip_example_scenarios). Having it in one repository makes it easier for end-user to use them for their work directly 
- [ ] Can you add the link to Contributor guidelines (mentioned in the [documentation](https://tfaip.readthedocs.io/en/latest/doc.development.html#guidelines)) in the main README.md file. [Sorry for asking you to do this 😅]

Concerns regarding the paper - 
- [ ] In the section "State of the Field," compare _tfaip_ with some other packages based on TensorFlow and provide advantages of tfaip over them. 
- [ ] Remove all the hyperlinks and add them in either the reference section or in footnotes. 

You can let me know if you have any arguments against any of the concerns I have raised. I am more than happy to discuss these. 😄 
I'll add more concerns as and when I find them!

**Comment by levimcclenny** [comment]
Hey @ChWick, I'm the second reviewer for JOSS. 

In addition to the above, I believe a short example in the text would be useful. This paper does something similar - https://joss.theoj.org/papers/10.21105/joss.03077

Also for comparison, I would list how this compares to things like AutoKeras, etc. so the users can see how this package fits into the AutoML pipeline. That would be useful for me (for example) as someone familiar, but not well-versed in the topic. I am interested in using AutoML in projects of my own and I believe a comparison or discussion about this packages capabilities vs. other state-of-the-art packages would be welcome so I can see how I may do so. 

Other than that the paper feels very abbreviated, but comparing to other JOSS papers I believe this is somewhat standard. Most others don't use bullets (from looking at a few of the recent ones on the page), so reconfiguring into a paragraph format could be good to make it more as a "journal paper" or application note style.

The software itself is pretty good, and very timely. I think the documentation is very professional. I agree that providing more about contribution guidelines would be welcome. You have how people can build a local version and change stuff, but no guidelines on how a user would submit a PR, for instance, or what you are interested in developing in the future for others to grab on to.

**Comment by ChWick** [comment]
@Het-Shah @levimcclenny Thank you so much for reviewing the paper, and - of course - the software! We really appreciate your efforts and time investments!
Also thanks for the two reasonable initial reviews! We will adopt the paper and repository accordingly. This will take some time, I will reply after everything is done!

Some specific questions @Het-Shah:
* What do you mean by "benchmark". Do you propose a "speed" or "accuracy" comparison to Tensorflow's MNIST-Tutorial, or both?
* Do you think that merging the "examples" and "base" tfaip repositories suffices to provide additional "tutorials", or do you recommend additional ones?

**Comment by Het-Shah** [comment]
> What do you mean by "benchmark". Do you propose a "speed" or "accuracy" comparison to Tensorflow's MNIST-Tutorial, or both?

I would personally prefer both and a mention of the hardware used (can be a GPU from colab as well). You can easily collect these metrics in one run, so shouldn't be a problem in my opinion. If you are already in process of running experiments just a comparison of validation accuracies will also suffice. 

>Do you think that merging the "examples" and "base" tfaip repositories suffices to provide additional "tutorials", or do you recommend additional ones?

I think merging those repositories is fine for this submission, but if in the future you want to add more examples it's totally up to you.

**Comment by ChWick** [comment]
Hi, we included you suggestions:

@Het-Shah 

* We added a brief benchmark in the README.md for MNIST and the data pipeline.
*  We merged the examples which is certainly beneficial!
*  We added [contribution guidelines](https://tfaip.readthedocs.io/en/latest/doc.development.html) to the docs.
*  We added a comparison to AutoML (do you have other suggestions for packages to compare to?)
*  We removed all optional hyperlinks.

@levimcclenny 
We included a paragraph about AutoML and reformatted the paper to be more "paper-like" (no bullet lists). You suggested to include a short example. Do you mean to include the [tutorial code](https://github.com/Planet-AI-GmbH/tfaip/tree/master/examples/tutorial/min) which is probably too long, or the "training calls" as seen in the [README.md](https://github.com/Planet-AI-GmbH/tfaip/blob/master/README.md)?

**Comment by Het-Shah** [comment]
Looks good to me!
